libs/JSystem/JKernel/JKRSolidHeap.cpp

```diff
@@ -1,112 +1,302 @@
 #include "JSystem/JKernel/JKRSolidHeap/JKRSolidHeap.h"
 #include "global.h"
 
-asm void JKRSolidHeap::create(unsigned long, JKRHeap*, bool) {
-    nofralloc
-#include "JSystem/JKernel/JKRSolidHeap/asm/func_802D0A24.s"
+JKRSolidHeap* JKRSolidHeap::create(u32 size, JKRHeap* heap, bool useErrorHandler) {
+    if (!heap) {
+        heap = getRootHeap();
+    }
+
+    if (size == 0xffffffff) {
+        size = heap->getMaxAllocatableSize(0x10);
+    }
+
+    u32 alignedSize = ALIGN_PREV(size, 0x10);
+    u32 solidHeapSize = ALIGN_NEXT(sizeof(JKRSolidHeap), 0x10);
+    if (alignedSize < solidHeapSize) {
```

Comment:
```suggestion
    if (alignedSize < solidHeapSize)
```
for consistency, or you should add parentheses around the condition below.